db/db_impl/db_impl_write.cc

```diff
@@ -2278,6 +2283,9 @@ size_t DBImpl::GetWalPreallocateBlockSize(uint64_t write_buffer_size) const {
   if (immutable_db_options_.write_buffer_manager) {
     size_t buffer_size =
         immutable_db_options_.write_buffer_manager->flush_size();
+    for (auto manager : cf_based_write_buffer_manager_) {
```

Comment:
Why do we need to update buffer_size for WAL here?

Reply:
If the flush size of write buffer manager is small compared with write buffer size, we will switch memtable before write bytes up to write buffer size. Use min here can avoid wasty pre allocation.

Reply:
but here you're using += instead of min<> of cf_based_write_buffer_manager_

Reply:
This is because all CF will write in the same WAL file. Imaging we have 1M write-buffer-limit for CF1 but 1GB write-buffer-limit for CF2.